src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -0,0 +1,547 @@
+                final byte refBase = ref[refOffset];
+                final RefVsAnyResult homRefCalc = calcGenotypeLikelihoodsOfRefVsAny(ploidy, pileup, refBase, BASE_QUAL_THRESHOLD, null);
+                homRefCalc.capByHomRefLikelihood();
+
+                final Allele refAllele = Allele.create(refBase, true);
+                final List<Allele> refSiteAlleles = Arrays.asList(refAllele, GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE);
+                final VariantContextBuilder vcb = new VariantContextBuilder("HC", curPos.getContig(), curPos.getStart(), curPos.getStart(), refSiteAlleles);
+                final GenotypeBuilder gb = new GenotypeBuilder(sampleName, GATKVariantContextUtils.homozygousAlleleList(refAllele, ploidy));
+                gb.AD(homRefCalc.get_AD_Ref_Any());
+                gb.DP(homRefCalc.getDP());
+
+                // genotype likelihood calculation
+                final GenotypeLikelihoods snpGLs = GenotypeLikelihoods.fromLog10Likelihoods(homRefCalc.getGenotypeLikelihoods());
+                final int nIndelInformativeReads = calcNIndelInformativeReads(pileup, refOffset, ref, indelInformativeDepthIndelSize);
+                final GenotypeLikelihoods indelGLs = getIndelPLs(ploidy,nIndelInformativeReads);
+
+                // now that we have the SNP and indel GLs, we take the one with the least confidence,
+                // as this is the most conservative estimate of our certainty that we are hom-ref.
+                // For example, if the SNP PLs are 0,10,100 and the indel PLs are 0,100,1000
+                // we are very certain that there's no indel here, but the SNP confidence imply that we are
+                // far less confident that the ref base is actually the only thing here.  So we take 0,10,100
+                // as our GLs for the site.
+                final GenotypeLikelihoods leastConfidenceGLs = getGLwithWorstGQ(indelGLs, snpGLs);
+
+                gb.GQ((int) (-10 * GenotypeLikelihoods.getGQLog10FromLikelihoods(GenotypeType.HOM_REF.ordinal()-1, leastConfidenceGLs.getAsVector())));
+                gb.PL(leastConfidenceGLs.getAsPLs());
+
+                vcb.genotypes(gb.make());
+                results.add(vcb.make());
+            }
+        }
+
+        return results;
+    }
+
+    /**
+     * Get the GenotypeLikelihoods with the least strong corresponding GQ value
+     * @param gl1 first to consider (cannot be null)
+     * @param gl2 second to consider (cannot be null)
+     * @return gl1 or gl2, whichever has the worst GQ
+     */
+    @VisibleForTesting
+    GenotypeLikelihoods getGLwithWorstGQ(final GenotypeLikelihoods gl1, final GenotypeLikelihoods gl2) {
+        return GenotypeLikelihoods.getGQLog10FromLikelihoods(GenotypeType.HOM_REF.ordinal()-1, gl1.getAsVector())
+                >
+               GenotypeLikelihoods.getGQLog10FromLikelihoods(GenotypeType.HOM_REF.ordinal()-1, gl2.getAsVector()) ? gl1 : gl2;
+    }
```

Comment:
Also can you make the code look nicer, perhaps using local variable for the large expressions.